server/db/driver/pg/orders.go

```diff
@@ -1285,30 +1217,50 @@ func loadTradeFromTable(dbe *sql.DB, fullTable string, oid order.OrderID) (order
 	return nil, 0, fmt.Errorf("unknown order type %d retrieved", prefix.OrderType)
 }
 
-func (a *Archiver) userOrders(ctx context.Context, base, quote uint32, aid account.AccountID) ([]order.Order, []pgOrderStatus, error) {
+// userOrders returns the orders with the specified order IDs (or all orders,
+// if no order ID is specified) for the given account in the market specified
+// by a base and quote asset.
+func (a *Archiver) userOrders(ctx context.Context, aid account.AccountID, base, quote uint32, oids []order.OrderID) ([]*db.Order, error) {
 	marketSchema, err := a.marketSchema(base, quote)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	// Active orders.
 	fullTable := fullOrderTableName(a.dbName, marketSchema, true)
-	orders, statuses, err := userOrdersFromTable(ctx, a.db, fullTable, base, quote, aid)
+	orders, err := userOrdersFromTable(ctx, a.db, fullTable, aid, oids)
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
-		return nil, nil, err
+		return nil, err
+	}
+
+	if len(oids) > 0 && len(oids) == len(orders) {
+		// found all requested orders
+		return orders, nil
+	}
```

Comment:
I'm not sure at all, but would it be possible for len to be the same but oids to differ?

Reply:
I don't think so, if len(oids) > 0, then those oids are used in the select query, so if equal-length orders are returned, it should match those oids.

Reply:
ok, that makes sense